packages/x-data-grid/src/hooks/utils/useGridSelector.ts

```diff
@@ -36,13 +36,16 @@ export const useGridSelector = <Api extends GridApiCommon, T>(
   apiRef: React.MutableRefObject<Api>,
   selector: ((state: Api['state']) => T) | OutputSelector<Api['state'], T>,
   equals: (a: T, b: T) => boolean = defaultCompare,
+  debounced: boolean = false,
 ) => {
   if (process.env.NODE_ENV !== 'production') {
     if (!apiRef.current.state) {
       stateNotInitializedWarning();
     }
   }
 
+  const debounceTimerRef = React.useRef<ReturnType<typeof setTimeout>>();
```

Comment:
I don't love the idea of using a timeout. Would there be a different way to solve the problem?

Reply:
Agree. Also UT are failing because they are not expecting synchronous calls. I'm checking if we have any other options.